probe/docker_process_mapper_test.go

```diff
@@ -0,0 +1,81 @@
+package main
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	docker "github.com/fsouza/go-dockerclient"
+)
+
+type mockDockerClient struct {
+	containers    []docker.APIContainers
+	containerInfo map[string]*docker.Container
+}
+
+func (m mockDockerClient) ListContainers(options docker.ListContainersOptions) ([]docker.APIContainers, error) {
+	return m.containers, nil
+}
+
+func (m mockDockerClient) InspectContainer(id string) (*docker.Container, error) {
+	return m.containerInfo[id], nil
+}
+
+func TestDockerProcessMapper(t *testing.T) {
+	oldPidTreeStub, oldDockerClientStub := newPidTreeStub, newDockerClient
+	defer func() {
+		newPidTreeStub = oldPidTreeStub
+		newDockerClient = oldDockerClientStub
+	}()
+
+	newPidTreeStub = func(procRoot string) (*pidTree, error) {
+		pid1 := &process{pid: 1}
+		pid2 := &process{pid: 2, ppid: 1, parent: pid1}
+		pid1.children = []*process{pid2}
+
+		return &pidTree{
+			processes: map[int]*process{
+				1: pid1, 2: pid2,
+			},
+		}, nil
+	}
+
+	newDockerClient = func(endpoint string) (dockerClient, error) {
+		return mockDockerClient{
+			containers: []docker.APIContainers{{ID: "foo"}},
+			containerInfo: map[string]*docker.Container{
+				"foo": {
+					ID:    "foo",
+					Name:  "bar",
+					State: docker.State{Pid: 1, Running: true},
+				},
+			},
+		}, nil
+	}
+
+	dockerMapper := newDockerMapper("/proc", 10*time.Second)
+	dockerIDMapper := dockerIDMapper{dockerMapper}
+	dockerNameMapper := dockerNameMapper{dockerMapper}
+
+	fmt.Printf("%+v", dockerMapper)
```

Comment:
Never fmt.Print or println in tests; use t.Logf or (if it's for debugging the test) just remove it.